src/main/java/io/vertx/core/http/HttpClient.java

```diff
@@ -1089,6 +1369,31 @@ WebSocketStream websocketStream(String requestURI, MultiMap headers, WebsocketVe
                                   String subProtocols);
 
   /**
+   * Set a redirect handler for the web client.
+   * <p>
+   * The redirect handler is called when a {@code 3xx} response is received and the request is configured to
+   * follow redirects with {@link HttpClientRequest#setFollowRedirects(boolean)}.
+   * <p>
+   * The redirect handler is passed the {@link HttpClientResponse}, it can return an {@link HttpClientRequest} or {@code null}.
+   * <ul>
+   *   <li>when null is returned, the original response is processed by the original request response handler</li>
+   *   <li>when a new {@code Future<HttpClientRequest>} is returned, the client will send this new request</li>
+   * </ul>
+   * The handler must return a {@code Future<HttpClientRequest>} unsent so the client can further configure it and send it.
+   *
+   * @param handler the new redirect handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  HttpClient redirectHandler(Function<HttpClientResponse, Future<HttpClientRequest>> handler);
+
+  /**
+   * @return the current redirect handler.
+   */
+  @GenIgnore
+  Function<HttpClientResponse, Future<HttpClientRequest>> redirectHandler();
```

Comment:
yes we do sometimes, it is here in case someone wants to sets its own handler but get the previous one to do some kind of decoration or only wants to handle a specific case and delegates the rest of the behavior to the previous handler